app/src/main/java/com/codingblocks/cbonlineapp/database/LibraryDao.kt

```diff
@@ -4,21 +4,29 @@ import androidx.lifecycle.LiveData
 import androidx.room.Dao
 import androidx.room.Query
 import com.codingblocks.cbonlineapp.database.models.BookmarkModel
+import com.codingblocks.cbonlineapp.database.models.CodeModel
 import com.codingblocks.cbonlineapp.database.models.ContentLecture
+import com.codingblocks.cbonlineapp.database.models.PdfModel
 
 /**
  * @author aggarwalpulkit596
  */
 @Dao
 interface LibraryDao {
     @Query("""
-        SELECT c.bookmarkUid,c.runAttemptId,c.contentId,c.sectionId,c.createdAt,b.title as contentName ,s.name as sectionName FROM  BookmarkModel c
+        SELECT c.bookmarkUid,c.runAttemptId,c.contentId,c.sectionId,c.createdAt,b.title as contentName ,s.name as sectionName, b.contentable  FROM  BookmarkModel c
         INNER JOIN ContentModel b ON b.ccid = c.contentId
  	   INNER JOIN SectionModel s ON s.csid = c.sectionId
        WHERE c.runAttemptId = :id ORDER BY c.createdAt DESC
     """)
     fun getBookmarks(id: String): LiveData<List<BookmarkModel>>
 
+    @Query("SELECT codeUid, codeContestId FROM ContentModel WHERE ccid = :contentId")
+    suspend fun getCodeChallenge(contentId: String): CodeModel
+
+    @Query("SELECT documentPdfLink, documentName FROM ContentModel WHERE ccid = :contentId")
```

Comment:
You've used live data as well as suspended function any particular reason for that ?

Reply:
CodeChallengeActivity have its viewmodel, so we can use suspend but PDFActivity doesn't have any viewModel. So I used livedata to fetch this.

Reply:
Pls suggest if their is any other way to get data from Activity rather than LiveData ??

Reply:
you can always use a suspended function I mean global scope exist everywhere,  infact i would prefer to use suspended function